src/core/lib/security/credentials/tls/grpc_tls_crl_provider.cc

```diff
@@ -0,0 +1,82 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include "src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h"
+
+#include <memory>
+#include <vector>
+
+#include <openssl/bio.h>
+#include <openssl/crypto.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include <grpc/support/log.h>
+
+namespace grpc_core {
+namespace experimental {
+
+CrlImpl::CrlImpl(X509_CRL* crl) : crl_(crl) {
+  char* buf = X509_NAME_oneline(X509_CRL_get_issuer(crl), nullptr, 0);
+  issuer_ = buf;
+  OPENSSL_free(buf);
+}
+
+CrlImpl::~CrlImpl() { X509_CRL_free(crl_); }
+
+std::string CrlImpl::Issuer() { return issuer_; }
+
+absl::StatusOr<std::unique_ptr<Crl>> Crl::Parse(absl::string_view crl_string) {
+  BIO* crl_bio =
```

Comment:
Please check that `crl_string.size() < INT_MAX` and, if not, return an invalid argument error.

Reply:
PTAL

Reply:
LGTM.